sdk/test/AzureTest/Scrubbing.swift

```diff
@@ -0,0 +1,120 @@
+        }
+        
+        guard var editedString = string else {
+            return nil
+        }
+        var range = NSRange(location: 0, length: editedString.utf8.count)
+        if Filter.subscriptionIDRegex.numberOfMatches(in: editedString, options: [], range: range) > 0 {
+            editedString = Filter.subscriptionIDRegex.stringByReplacingMatches(in: editedString, options: [], range: range, withTemplate: "$1/99999999-9999-9999-9999-999999999999")
+        }
+        range = NSRange(location: 0, length: editedString.utf8.count)
+        if Filter.graphSubscriptionIDRegex.numberOfMatches(in: editedString, options: [], range: range) > 0 {
+            editedString = Filter.graphSubscriptionIDRegex.stringByReplacingMatches(in: editedString, options: [], range: range, withTemplate: "$1/99999999-9999-9999-9999-999999999999")
+        }
+        
+        for key in embeddedStringKeys {
+            let embeddedRegex = embeddedIDRegex(key)
+            range = NSRange(location: 0, length: editedString.utf8.count)
+            if embeddedRegex.numberOfMatches(in: editedString, options: [], range: range) > 0 {
+                editedString = embeddedRegex.stringByReplacingMatches(in: editedString, options: [], range: range, withTemplate: "\"\(key)\":\"/99999999-9999-9999-9999-999999999999\"")
+            }
+        }
+        return editedString
+        
+        
+    }
+    
+    static func scrubSubscriptionIDs(request: URLRequest) -> URLRequest {
+        
+        var cleanRequest = request
+        
+        let dirtyHeaders = request.allHTTPHeaderFields ?? [:]
+        let cleanHeaders = dirtyHeaders.mapValues { val -> String in
+            guard let newValue = scrubSubscriptionId(from: val) else {
+                return val
+            }
+            return newValue
+        }
+        cleanRequest.allHTTPHeaderFields = cleanHeaders.isEmpty ? nil : cleanHeaders
+        
+        cleanRequest.url = URL(string: scrubSubscriptionId(from: request.url?.absoluteString))
+        cleanRequest.mainDocumentURL = URL(string: scrubSubscriptionId(from: request.mainDocumentURL?.absoluteString))
+        cleanRequest.httpBody = scrubSubscriptionId(from: String(data: request.httpBody, encoding: .utf8))?.data(using: .utf8)
+
+        return cleanRequest
+    }
+    
+    static func scrubSubscriptionIDs(response: Foundation.URLResponse, data: Data?) -> (Foundation.URLResponse, Data?) {
+        guard let dirtyData = data else {
+            return (response, data)
+        }
+        
+        do {
+            guard let responseDataString = try JSONSerialization.jsonObject(with: dirtyData, options: .allowFragments) as? String else {
+                return (response, data)
+            }
+            let cleanedDataString = scrubSubscriptionId(from: responseDataString)
+            let cleanedData = cleanedDataString?.data(using: .utf8)
+            return (response, cleanedData)
```

Comment:
You could make this more concise. However, I would question whether it makes sense to use `?` here. If `scrubSubscriptionId` returns nil for some weird reason, it will `nil` out the data that wouldn't otherwise be `nil`.

```suggestion
            let cleanedData = scrubSubscriptionId(from: responseDataString)?.data(using: .utf8)
            return (response, cleanedData)
```